### PR TITLE
added container for emmtyper

### PR DESCRIPTION
**Pull request by hkunerth**
Description: added container for emmtyper, a program to assign emm-types to group A strep datasets. This is my second Docker build (just finished the workshop) and first using a micromamba base, so I'd be very happy to receive feedback!

The README is a bit over 30 lines, as there was no paper listed for citation, so I copied over the contributor and maintainer info from the emmtyper README. It can be removed or shortened if needed.

Pull Request (PR) checklist:
- [ x] Include a description of what is in this pull request in this message.
- [x ] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [ x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [ ] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [ x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x ] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x ] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x ] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x ] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: Program_Licenses.md [hunk 1]
```diff
@@ -27,8 +27,9 @@ The licenses of the open-source software that is contained in these Docker image
 | datasets-sars-cov-2 | Apache 2.0 | https://github.com/CDCgov/datasets-sars-cov-2/blob/master/LICENSE |
 | dragonflye | GNU GPLv3 | https://github.com/rpetit3/dragonflye/blob/main/LICENSE |
 | DSK | GNU Affero GPLv3 | https://github.com/GATB/dsk/blob/master/LICENSE |
-| emm-typing-tool | GNU GPLv3 | https://github.com/phe-bioinformatics/emm-typing-tool/blob/master/LICENCE |
 | emboss | GNU GPLv3 | http://emboss.sourceforge.net/licence/ |
+| emmtyper | GNU GPLv3 | https://github.com/MDU-PHL/emmtyper/blob/master/LICENSE |
```
Comment (erinyoung) [inline Program_Licenses.md:31]: Good catch on the alphabetical error!

File: emmtyper/0.2.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# set the environment
+ENV PATH="$PATH" \
+ LC_ALL=C.UTF-8
+
+#install emmtyper and dependencies
+RUN micromamba create -n emmtyper && \
+    eval "$(micromamba shell hook --shell=bash)" && \
+    micromamba activate emmtyper && \
+    micromamba install --yes --name emmtyper -c conda-forge -c bioconda -c defaults emmtyper && \
+    micromamba clean --all --yes
```
Comment (erinyoung) [inline emmtyper/0.2.0/Dockerfile:35]: Could you specify which emmtyper version you're using? It helps with the longevity of the Dockerfiles and images.

An example you could use is [Mykrobe](https://github.com/StaPH-B/docker-builds/blob/master/mykrobe/0.12.1/Dockerfile#L64-L69)

```
RUN micromamba create -n mykrobe -y -c conda-forge -c bioconda -c defaults \
 mykrobe=${MYKROBE_VER} \
 python \ 
 pip \
 pandas && \
 micromamba clean -a -y
 ```

File: emmtyper/0.2.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+ENV ENV_NAME="emmtyper"
+ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+RUN wget 'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/006/785/GCA_000006785.2_ASM678v2/GCA_000006785.2_ASM678v2_genomic.fna.gz' && \
+	gunzip GCA_000006785.2_ASM678v2_genomic.fna.gz && \
+	mv GCA_000006785.2_ASM678v2_genomic.fna test_data.fasta
+RUN emmtyper test_data.fasta && \
+	emmtyper -w pcr test_data.fasta -o test_out && \
+	head -10 test_out
```
Comment (erinyoung) [inline emmtyper/0.2.0/Dockerfile:51]: I like this test : downloads a fasta file and then runs emmtyper on it.

**Review by erinyoung — CHANGES_REQUESTED** [review]
Looks great. Here are some changes that need to be made
- version is added when installing emmtyper because it helps with stability
- Right now the testing phase fails because `/usr/local/bin/_entrypoint.sh: line 24: exec: ps: not found` - this is remedied by installing `procps` via apt-get.

Something like:
```
RUN apt-get update && apt-get install -y --no-install-recommends \
 wget \
 procps && \
 apt-get autoclean && rm -rf /var/lib/apt/lists/*
```

File: emmtyper/0.2.0/Dockerfile (hunk 2 above)
Comment (hkunerth) [reply emmtyper/0.2.0/Dockerfile]: Makes sense - I followed the Mykrobe approach in the updated version.

File: emmtyper/0.2.0/Dockerfile (hunk 3 above)
Comment (hkunerth) [reply emmtyper/0.2.0/Dockerfile:53]: Getting the isPCR version to work within the container was a big headache, which is why I pivoted to a micromamba / conda install. Worked fine after that!

**Comment by hkunerth** [comment]
I've completed the changes and my build-to-test on Github actions was a success (though my test prior to the pull request worked as well). I have added procps, so hopefully we are all set now.

**Comment by hkunerth** [comment]
Certificates issue.. trying again.

**Comment by erinyoung** [comment]
Sorry for not catching the lack of ca-certificates sooner! As background for this : ps and ca-certificates are used by nextflow-tower, so we required them in our images, but they aren't normally required for Dockerfiles.

**Comment by hkunerth** [comment]
> Sorry for not catching the lack of ca-certificates sooner! As background for this : ps and ca-certificates are used by nextflow-tower, so we required them in our images, but they aren't normally required for Dockerfiles.

No worries, and that's great to know! I was wondering about why they were so ubiquitous.

**Review by erinyoung — APPROVED** [review]
Looks good!

**Comment by erinyoung** [comment]
I've merged this PR, and I am currently deploying the image to both Dockerhub and Quay. This should be done _soon_. You can watch the progress here : https://github.com/StaPH-B/docker-builds/actions/runs/4128510659

**Pull request merged**